packages/core/src/types.ts

```diff
@@ -19,55 +57,213 @@ export interface IV1Wallet {
+   * @returns {Promise<WalletDocument[]>} Array of all documents in the wallet
+   */
   getAllDocuments: () => Promise<WalletDocument[]>;
+  
+  /**
+   * Adds a new document to the wallet
+   * @param {any} json - The document to add (must have valid JSON-LD structure)
+   * @param {any} [options] - Optional parameters for document creation
+   * @returns {Promise<WalletDocument>} The created document with generated metadata
+   * @fires WalletEvents.documentAdded
+   */
   addDocument: (json: any, options?: any) => Promise<WalletDocument>;
+  
+  /**
+   * Creates or updates a document based on its ID
+   * @param {any} json - The document to create or update
+   * @param {any} [options] - Optional parameters for the operation
+   * @returns {Promise<WalletDocument>} The created or updated document
+   */
   upsertDocument: (json: any, options?: any) => Promise<WalletDocument>;
+  
+  /**
+   * Updates an existing document
+   * @param {any} json - The document with updated data (must include ID)
+   * @param {any} [options] - Optional parameters for document update
+   * @returns {Promise<WalletDocument>} The updated document
+   * @fires WalletEvents.documentUpdated
+   * @throws {Error} If document doesn't exist
+   */
   updateDocument: (json: any, options?: any) => Promise<WalletDocument>;
+  
+  /**
+   * Gets all documents correlated to a specific document
+   * @param {string} documentId - The ID of the document to find correlations for
+   * @returns {Promise<WalletDocument[]>} Array of correlated documents
+   */
   getDocumentCorrelations: (documentId: string) => Promise<WalletDocument[]>;
+  
+  /**
+   * Retrieves the keypair associated with an account
+   * @param {string} accountId - The account ID to get the keypair for
+   * @returns {Promise<any>} The keypair associated with the account
+   */
   getAccountKeyPair: (accountId: string) => Promise<any>;
+  
   /**
-   * Remove document by id
-   * @param id
+   * Removes a document from the wallet
+   * @param {string} id - The ID of the document to remove
+   * @param {any} [options] - Optional parameters for document removal
+   * @returns {Promise<void>}
+   * @fires WalletEvents.documentRemoved
+   * @throws {Error} If document is not found
    */
   removeDocument: (id: string, options?: any) => Promise<void>;
+  
   /**
-   * Import data from a Universal Wallet 2020 JSON
-   * https://w3c-ccg.github.io/universal-wallet-interop-spec/
-   *
-   * @param json
+   * Imports documents from an encrypted Universal Wallet 2020 JSON
+   * @param {any} json - The encrypted wallet JSON
+   * @param {string} password - Password to decrypt the wallet
+   * @returns {Promise<void>}
+   * @see {@link https://w3c-ccg.github.io/universal-wallet-interop-spec/}
    */
   importUniversalWalletJSON: (json: any, password: string) => Promise<void>;
+  
+  /**
+   * Decrypts and retrieves documents from an encrypted wallet without importing
+   * @param {any} json - The encrypted wallet JSON
```

Comment:
is this also expecting the universal wallet format? Should we add a `@see` reference here too?